src/server.rs

```diff
@@ -448,6 +451,7 @@ fn do_update_loop<C: Config>(server: &mut Server<C>) -> ShutdownResult {
         });
 
         server.player_lists.update();
+        server.inventories.sync(&mut server.clients);
```

Comment:
The problem here is that this function sends packets to clients, but the client packets have already been flushed. I think the work of this function that sends packets to clients should be done in the client's `update()` function near where the client's own inventory is updated. Then this function can be renamed to `update()` and will only have to mark the inventories as clean.